src/internal/operators/bufferToggle.ts

```diff
@@ -56,123 +58,56 @@ export function bufferToggle<T, O>(
   closingSelector: (value: O) => SubscribableOrPromise<any>
 ): OperatorFunction<T, T[]> {
   return function bufferToggleOperatorFunction(source: Observable<T>) {
-    return lift(source, new BufferToggleOperator<T, O>(openings, closingSelector));
+    return wrappedLift(source, (subscriber, liftedSource) => {
+      const buffers: T[][] = [];
+
+      // Subscribe to the openings notifier first
+      from(openings).subscribe(
+        new OperatorSubscriber(
+          subscriber,
+          (openValue) => {
+            const buffer: T[] = [];
+            buffers.push(buffer);
+            // We use this composite subscription, so that
+            // when the closing notifier emits, we can tear it down.
+            const closingSubscription = new Subscription();
+
+            // This is captured here, because we emit on both next or
+            // if the closing notifier completes without value.
+            // TODO: We probably want to not have closing notifiers emit!!
+            const emit = () => {
+              arrRemove(buffers, buffer);
+              subscriber.next(buffer);
+              closingSubscription.unsubscribe();
+            };
+
+            // The line below will add the subscription to the parent subscriber *and* the closing subscription.
+            closingSubscription.add(from(closingSelector(openValue)).subscribe(new OperatorSubscriber(subscriber, emit, undefined, emit)));
+          },
+          undefined,
+          noop
```

Comment:
nitpick/question: why is a `noop` needed here? The `onError` and `onComplete` parameter are optional, AFAICT, so why not omit them?